### PR TITLE
Add docs for the scheduling.cast.ai/storage-optimized label

### PR DESCRIPTION
**Pull request by noelruault**
Adds a new reference page that describes how to configure node affinity for storage-optimized pod/pods, nodes and instances.

### CODE REVIEW

File: docs/guides/storage-optimized.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+description: CAST AI supports running your workloads on Storage Optimized instances. This guide helps you configure and run it.
+---
+
+# Storage Optimized Instances
+
+The CAST AI autoscaler supports running your workloads on Storage Optimized instances.
+This guide will help you configure and run it.
+
+## Available configurations
```
Comment (viktorasm) [inline docs/guides/storage-optimized.md:10]: Before this, explain storage optimized instances better.

Mention:
* what do you mean by storage optimized: explain local disks vs network storage, maybe enough to link to provider specific docs
* is it more expensive? should I use it every time? what are restrictions (less instance families will be available, more expensive etc)
* how is it different from cloud to cloud?
* how to trigger provision: what triggers creation of storage optimized instance ( i guess it's the node selector)
* differences with "normal" nodes - e.g. node contains a taint, preventing workloads without special taints to run there

Comment (noelruault) [reply docs/guides/storage-optimized.md:35]: I have added an extended description and a reference to the official docs and pricing. Please let me know what you think.

File: docs/guides/storage-optimized.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- GCS: [Local SSD disks](https://cloud.google.com/compute/docs/disks#localssds)
+
+The CAST AI autoscaler supports running your workloads on Storage Optimized instances.
+The following guide will help you configure and run it.
+
+## Pricing
+
+### GCS
+
+Because local SSDs can only be purchased in 375 GB increments, the cost-per-month for a single device is the monthly rate multiplied by 375 GB. For example, at a monthly rate of $0.080, the cost would be $30.00 per device per month. Actual data storage and usage are included in that price and there is no additional charge for local traffic between the virtual machine and the local SSD device. [Local SSD prices](https://cloud.google.com/compute/disks-image-pricing#localssdpricing) differ by region.
```
Comment (viktorasm) [inline docs/guides/storage-optimized.md:23]: "monthly rate of $0.080" - is this per GB? maybe "monthly rate of $0.080/GB" then?

File: docs/guides/storage-optimized.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+```yaml
+tolerations:
+  - key: scheduling.cast.ai/storage-optimized
+    operator: Exists
+```
+
+### Node Selectors
+
+**When to use:** only use Storage Optimized instances
```
Comment (viktorasm) [inline docs/guides/storage-optimized.md:53]: not sure if we need these two cases. it's an added cost, so most probably users need second case most of the time.

I'd just explain "selector+toleration" as main use case, and add a second chapter to explain how node creation will be handled if there's no selector, just toleration. E.g. if I have only one pending pod with just the toleration, will the pod get a optimized node or not (probably it's not)

Comment (noelruault) [reply docs/guides/storage-optimized.md:41]: Hi Viktoras, thanks for your input so far. 
I'm wondering if you are thinking of this second chapter as a link that forwards you to another doc, probably mentioned at the bottom of the page? As it sounds, it seems that we could refer to some generic page that describes how node affinity is managed by CAST AI. This way we could use it for other similar guides (like Spot). 
What do you think?

**Pull request closed without merging**